src/Microsoft.Health.Fhir.Core.UnitTests/Features/Operations/Import/ImportOrchestratorJobTests.cs

```diff
@@ -68,16 +86,561 @@ public async Task GivenAnOrchestratorJobAndWrongEtag_WhenOrchestratorJobStart_Th
+
+            ImportOrchestratorJob orchestratorJob = new ImportOrchestratorJob(
+                mediator,
+                importOrchestratorInputData,
+                new ImportOrchestratorJobResult(),
+                contextAccessor,
+                fhirDataBulkImportOperation,
+                integrationDataStoreClient,
+                testQueueClient,
+                orchestratorJobInfo,
+                new Configs.ImportTaskConfiguration() { MaxRunningProcessingJobCount = 1 },
+                loggerFactory);
+            orchestratorJob.PollingFrequencyInSeconds = 0;
+
+            await Assert.ThrowsAnyAsync<RetriableJobException>(() => orchestratorJob.ExecuteAsync(new Progress<string>(), CancellationToken.None));
+
+            _ = mediator.DidNotReceive().Publish(
+                Arg.Any<ImportJobMetricsNotification>(),
+                Arg.Any<CancellationToken>());
+        }
+
+        [Fact]
+        public async Task GivenAnOrchestratorJob_WhenLastSubJobFailed_ThenImportProcessingExceptionShouldBeThrowAndWaitForOtherSubJobsCompleted()
+        {
+            IImportOrchestratorJobDataStoreOperation fhirDataBulkImportOperation = Substitute.For<IImportOrchestratorJobDataStoreOperation>();
+            RequestContextAccessor<IFhirRequestContext> contextAccessor = Substitute.For<RequestContextAccessor<IFhirRequestContext>>();
+            ILoggerFactory loggerFactory = new NullLoggerFactory();
+            IIntegrationDataStoreClient integrationDataStoreClient = Substitute.For<IIntegrationDataStoreClient>();
+            ISequenceIdGenerator<long> sequenceIdGenerator = Substitute.For<ISequenceIdGenerator<long>>();
+            IMediator mediator = Substitute.For<IMediator>();
+            ImportOrchestratorJobInputData importOrchestratorInputData = new ImportOrchestratorJobInputData();
+            ImportOrchestratorJobResult importOrchestratorJobResult = new ImportOrchestratorJobResult();
+            TestQueueClient testQueueClient = new TestQueueClient();
+            int getJobByGroupIdCalledTime = 0;
+
+            testQueueClient.GetJobByIdFunc = (queueClient, id, _) =>
+            {
+                JobInfo jobInfo = queueClient.JobInfos.First(t => t.Id == id);
+                if (jobInfo.Id == 3)
+                {
+                    jobInfo.Status = JobStatus.Failed;
+                    jobInfo.Result = JsonConvert.SerializeObject(new ImportProcessingJobErrorResult() { Message = "Job Failed"});
+                }
+
+                return jobInfo;
+            };
+            testQueueClient.GetJobByGroupIdFunc = (queueClient, groupId, _) =>
+            {
+                IEnumerable<JobInfo> jobInfos = queueClient.JobInfos.Where(t => t.GroupId == groupId);
+                if (++getJobByGroupIdCalledTime > 1)
```

Comment:
>getJobByGroupIdCalledTime [](http://example.com/codeflow?start=22&length=25)

For the first time, it is 0? Or getJobByGroupIdCalledTime++?

Reply:
Yes, I set it as 0 at line 336.